### PR TITLE
Settings: Fix stories archives dropdown search input

### LINKED ISSUE #13280: Stories Archives Create Your Own Select Page Search Field Wonkiness
## Bug Description
Noticed the following on Stories Archives:

1. Clicking on the search field for Stories Archive - Create Your Own setting does not populate the available pages to select as the custom archive page. 

2. Clicking "x" does not remove the value in the search field.

3. Balloon for available pages to select floats to the top of the setting page.

4. HTML characters displaying in search results for page that has them.

## Expected Behaviour

- Search icon should populate pages to set as custom archive page. 
- Typing in the URL in the search field should populate page to select directly under the field.

## Steps to Reproduce

1. Go to Settings > Stories Archives
2. Select Create Your Own
3. Click on Search Icon or type in URL
4. Notice 0 results populate 
5. Select a page 
6. click x button - notice inaction

## Screenshots

https://github.com/GoogleForCreators/web-stories-wp/assets/66372350/e40e0380-84e6-4036-baf3-14b04d054dda

https://github.com/GoogleForCreators/web-stories-wp/assets/66372350/68aaed3a-6130-453f-8533-9d974be1a051

Screenshot 2023-05-09 at 9 29 08 AM

https://github.com/GoogleForCreators/web-stories-wp/assets/66372350/42b72f6b-99d3-42b4-9db1-3a679538059d

## Additional Context

- Plugin Version: 1.32.0
- WordPress Version:
- Operating System:
- Browser:

### PR DESCRIPTION
**Pull request by AnuragVasanwala**
## Context

The goal of the PR is to fix https://github.com/GoogleForCreators/web-stories-wp/issues/13280.

Additionally, it also fixes SearchBox Cheveron Icon that matches with other input controls on the same page.

## Summary

This PR addresses issues related with the drop-down button (chevron up-down) and clear icon 'x'. As per this PR, new user flow would be:

| Scenario         | Image / Animated GIF |
| ------------- | ------------- |
| Clicking on the Chevron/changing text will open the popup below search box. | image |
| Clicking on Chevron (when search-box value is not edited) will toggle icon and popup. | toggle-popup |
| Changing text will hide Chevron and show 'X' clear button. | image |
| Clicking on the 'X' clear button will clear out search text. | clear-searchbox |

## To-do

Not applicable.

## User-facing changes

| Old SVG Icon         | Updated SVG Icon |
| ------------- | ------------- |
| Having different size (actual SVG icons are different).  | Corrected with the same SVG icons used on other input element as seen on the same page.  |
| Screenshot 2023-05-17 at 2 32 12 PM  | Screenshot 2023-05-17 at 2 28 53 PM  |

## Testing Instructions

Please refer [issues:13280](https://github.com/GoogleForCreators/web-stories-wp/issues/13280) for detailed testing instruction.

### Does this PR have a security-related impact?

No.

### Does this PR change what data or activity we track or use?

No.

### Does this PR have a legal-related impact?

No.

## Checklist

- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes https://github.com/GoogleForCreators/web-stories-wp/issues/13280

### CODE REVIEW

**Comment by AnuragVasanwala** [comment]
@swissspidy Would you please look into failed checks and let me know how should I resolve them?

**Comment by swissspidy** [comment]
11e1f72b7d565f02b77458cf05a3d5c8b506a7f8 hopefully helps. Otherwise apparently it'll get fixed in Bun this week.

File: packages/design-system/src/components/search/input/input.tsx [hunk 1]
```diff
@@ -82,12 +89,29 @@ const SearchInput = forwardRef(
...
+        if (value === '') {
+          setIsEmptyValue(true);
+        } else {
+          setIsEmptyValue(false);
+        }
+        if (isOpen) {
+          setInputChanged(true);
+        } else {
+          setInputChanged(false);
+        }
```
Comment (swissspidy) [inline packages/design-system/src/components/search/input/input.tsx]:
```suggestion
        setIsEmptyValue(value === '');
        setInputChanged(isOpen);
```

**Review by swissspidy — APPROVED** [review]
Just a tiny remark, otherwise this looks great!

Thanks a lot for the thought put into changing the chevron icon and the videos/screenshots to make the review super easy!

**Pull request merged**